packages/container/libraries/core/src/metadata/actions/updateMaybeClassMetadataPostConstructor.ts

```diff
@@ -1,19 +1,10 @@
-import { InversifyCoreError } from '../../error/models/InversifyCoreError';
-import { InversifyCoreErrorKind } from '../../error/models/InversifyCoreErrorKind';
 import { MaybeClassMetadata } from '../models/MaybeClassMetadata';
 
 export function updateMaybeClassMetadataPostConstructor(
   methodName: string | symbol,
 ): (metadata: MaybeClassMetadata) => MaybeClassMetadata {
   return (metadata: MaybeClassMetadata): MaybeClassMetadata => {
-    if (metadata.lifecycle.postConstructMethodName !== undefined) {
-      throw new InversifyCoreError(
-        InversifyCoreErrorKind.injectionDecoratorConflict,
-        'Unexpected duplicated postConstruct decorator',
-      );
-    }
-
-    metadata.lifecycle.postConstructMethodName = methodName;
+    metadata.lifecycle.postConstructMethodNames.push(methodName);
```

Comment:
Throw error in case `methodName` is already in `metadata.lifecycle.postConstructMethodNames`.